src/nbl/video/CVulkanConnection.h

```diff
@@ -59,21 +99,54 @@ class CVulkanConnection final : public IAPIConnection
         return core::SRange<const core::smart_refctd_ptr<IPhysicalDevice>>{ m_physDevices->begin(), m_physDevices->end() };
     }
 
+    core::smart_refctd_ptr<ISurface> createSurface(ui::IWindow* window) const override;
+
     VkInstance getInternalObject() const { return m_instance; }
 
 protected:
     ~CVulkanConnection()
     {
+        vkDestroyDebugUtilsMessengerEXT(m_instance, m_debugMessenger, nullptr);
         vkDestroyInstance(m_instance, nullptr);
     }
 
 private:
-    VkInstance m_instance = nullptr;
+    VkInstance m_instance = VK_NULL_HANDLE;
+    VkDebugUtilsMessengerEXT m_debugMessenger = VK_NULL_HANDLE;
     using physical_devs_array_t = core::smart_refctd_dynamic_array<core::smart_refctd_ptr<IPhysicalDevice>>;
     physical_devs_array_t m_physDevices;
+
+    static VKAPI_ATTR VkBool32 VKAPI_CALL placeholderDebugCallback(VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity, VkDebugUtilsMessageTypeFlagsEXT messageType,
+        const VkDebugUtilsMessengerCallbackDataEXT* callbackData, void* userData)
+    {
+        printf("Validation Layer: %s\n", callbackData->pMessage);
+        return VK_FALSE;
+    }
+
+    static inline bool areAllInstanceLayersAvailable(const std::vector<const char*>& requiredInstanceLayerNames)
+    {
+        uint32_t instanceLayerCount;
+        vkEnumerateInstanceLayerProperties(&instanceLayerCount, nullptr);
+        std::vector<VkLayerProperties> instanceLayers(instanceLayerCount);
+        vkEnumerateInstanceLayerProperties(&instanceLayerCount, instanceLayers.data());
+
+        for (const auto& requiredLayerName : requiredInstanceLayerNames)
+        {
+            const auto& result = std::find_if(instanceLayers.begin(), instanceLayers.end(),
+                [requiredLayerName](const VkLayerProperties& layer) -> bool
+                {
+                    return (strcmp(requiredLayerName, layer.layerName) == 0);
+                });
+
+            if (result == instanceLayers.end())
+                return false;
+        }
```

Comment:
there's probably an STL thing for this (verifing if all objects from one iterator range are present in another iterator range)